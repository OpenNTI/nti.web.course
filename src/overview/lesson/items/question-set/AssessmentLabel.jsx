import './AssessmentLabel.scss';
import React from 'react';
import PropTypes from 'prop-types';
import { scoped } from '@nti/lib-locale';
import { List } from '@nti/web-commons';

import Required from '../../common/Required';
import RequirementControl from '../../../../progress/widgets/RequirementControl';

const t = scoped('course.overview.lesson.items.question-set.List', {
	questionCount: {
		one: '%(count)s Question',
		other: '%(count)s Questions',
	},
	correct: '%(correct)s Correct',
	incorrect: '%(incorrect)s Incorrect',
});

LessonOverviewQuestionSetAssessmentLabel.propTypes = {
	overviewItemRef: PropTypes.object,
	assessment: PropTypes.object,
	assessmentSubmission: PropTypes.object,
	required: PropTypes.bool,
	onRequirementChange: PropTypes.func,
	editMode: PropTypes.bool,
};
export default function LessonOverviewQuestionSetAssessmentLabel({
	assessment,
	assessmentSubmission,
	required,
	overviewItemRef,
	onRequirementChange,
}) {
	const count = assessment ? parseInt(assessment['question-count'], 10) : 0;
	const correct = assessmentSubmission && assessmentSubmission.getCorrect();
	const incorrect =
		assessmentSubmission && assessmentSubmission.getIncorrect();

	const requiredWidget =
		onRequirementChange &&
		overviewItemRef.isCompletable &&
		overviewItemRef.isCompletable() ? (
			<RequirementControl
				record={overviewItemRef}
				onChange={onRequirementChange}
			/>
		) : (
			required && <Required />
		);

	return (
		<List.SeparatedInline className="lesson-overview-questionset-assessment-label">
			{requiredWidget}
			{!assessmentSubmission && (
				<span className="question-count">
					{t('questionCount', { count })}
				</span>
			)}
			{assessmentSubmission && (
				<div className="submitted">
					<span className="correct">
						<i className="icon-check-10" />
						<span>{t('correct', { correct })}</span>
					</span>
					<span className="incorrect">
						<i className="icon-bold-x" />
						<span>{t('incorrect', { incorrect })}</span>
					</span>
				</div>
			)}
		</List.SeparatedInline>
	);
}
