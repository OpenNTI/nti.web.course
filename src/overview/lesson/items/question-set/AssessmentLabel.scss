@import '~@nti/style-common';

.lesson-overview-questionset-assessment-label {
	font: normal 400 0.75rem/1.4 $body-font-family;
	color: $tertiary-grey;

	.require-control-value {
		text-transform: uppercase;
	}

	.submitted {
		.correct, .incorrect {
			i, span {
				display: inline-block;
				vertical-align: middle;
			}

			i {
				font-size: 1.125rem;
				margin-right: rem-calc(2px);
			}
		}

		.correct {
			color: $primary-green;
			margin-right: 1rem;
		}

		.incorrect {
			color: $primary-red;
		}
	}
}
