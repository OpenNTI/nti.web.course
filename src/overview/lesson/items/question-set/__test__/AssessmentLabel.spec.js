import React from 'react';
import renderer from 'react-test-renderer';

import AssessmentLabel from '../AssessmentLabel';

/* eslint-env jest */
describe('Course overview asessment label test', () => {
	test('Question count', async () => {
		const assessment = {
			'question-count': 15
		};

		const cmp = renderer.create(
			<AssessmentLabel assessment={assessment}/>
		);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Question count with correct/incorrect', async () => {
		const assessment = {
			'question-count': 15
		};

		const assessmentSubmission = {
			getCorrect: () => 5,
			getIncorrect: () => 10
		};

		const cmp = renderer.create(
			<AssessmentLabel assessment={assessment} assessmentSubmission={assessmentSubmission}/>
		);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Question count, required', async () => {
		const assessment = {
			'question-count': 15
		};

		const cmp = renderer.create(
			<AssessmentLabel assessment={assessment} required/>
		);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});

	test('Question count, required editable', async () => {
		const assessment = {
			'question-count': 15
		};

		const overviewItemRef = {
			isCompletable: () => true
		};

		const cmp = renderer.create(
			<AssessmentLabel assessment={assessment} overviewItemRef={overviewItemRef} onRequirementChange={()=>{}}/>
		);

		const tree = cmp.toJSON();

		expect(tree).toMatchSnapshot();
	});
});
