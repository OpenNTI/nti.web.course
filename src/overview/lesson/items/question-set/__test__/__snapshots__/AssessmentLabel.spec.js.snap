// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course overview asessment label test Question count 1`] = `
<ul
  className="nti-separated-inline-list lesson-overview-questionset-assessment-label"
>
  <li>
    <span
      className="question-count"
    >
      15 Questions
    </span>
  </li>
</ul>
`;

exports[`Course overview asessment label test Question count with correct/incorrect 1`] = `
<ul
  className="nti-separated-inline-list lesson-overview-questionset-assessment-label"
>
  <li>
    <div
      className="submitted"
    >
      <span
        className="correct"
      >
        <i
          className="icon-check-10"
        />
        <span>
          5 Correct
        </span>
      </span>
      <span
        className="incorrect"
      >
        <i
          className="icon-bold-x"
        />
        <span>
          10 Incorrect
        </span>
      </span>
    </div>
  </li>
</ul>
`;

exports[`Course overview asessment label test Question count, required 1`] = `
<ul
  className="nti-separated-inline-list lesson-overview-questionset-assessment-label"
>
  <li>
    <div
      className="lesson-overview-item-required-label"
    >
      Required
    </div>
  </li>
  <li>
    <span
      className="question-count"
    >
      15 Questions
    </span>
  </li>
</ul>
`;

exports[`Course overview asessment label test Question count, required editable 1`] = `
<ul
  className="nti-separated-inline-list lesson-overview-questionset-assessment-label"
>
  <li>
    <div
      aria-haspopup="dialog"
      className="flyout-trigger require-control-value require-control flyout-closed"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      tabIndex={0}
    >
      <span>
        Optional
      </span>
      <i
        className="icon-chevron-down"
      />
    </div>
  </li>
  <li>
    <span
      className="question-count"
    >
      15 Questions
    </span>
  </li>
</ul>
`;
